### PR TITLE
Remove unnecessary merging from child context to the context

### PR DESCRIPTION
**Pull request by pitiphong-p**
CoreData implementation will push the saved changes from child context to the parent context when save.

So I think the merge change from childContext to the context code is unnecessary and also make the leaked on the observer (addObserverForName:object:queue:usingBlock: return the observer object for remove it from the Notification Center later)

Please read: "Core Data Release Notes for iOS v5.0" documentation for more information.

### CODE REVIEW

File: AFIncrementalStore/AFIncrementalStore.m [hunk 1]
```diff
@@ -524,10 +524,6 @@ - (id)newValueForRelationship:(NSRelationshipDescription *)relationship
             NSManagedObjectContext *childContext = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
             childContext.parentContext = context;
             childContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy;
-                        
-            [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:childContext queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *note) {
```
Comment (pitiphong-p) [inline AFIncrementalStore/AFIncrementalStore.m:528]: The addObserverForName:object:queue:usingBlock: will return the observer object. We must use it to remove it from observing later.

File: AFIncrementalStore/AFIncrementalStore.m [hunk 2]
```diff
@@ -524,10 +524,6 @@ - (id)newValueForRelationship:(NSRelationshipDescription *)relationship
...
-                [context mergeChangesFromContextDidSaveNotification:note];
```
Comment (pitiphong-p) [inline AFIncrementalStore/AFIncrementalStore.m:529]: Child context will push the changes to the parent context when save. So we don't need to merge changes from child context again.

**Comment by mattt** [comment]
@pitiphong-p I've read the release notes extensively and watched the WWDC sessions many times, but in my experience from developing this originally, I found that the child context was, in fact, not actually notifying its parent context of its saves. Now, this could have been caused by an entirely orthogonal mistake on my part, that has since been made moot. Either way, I'll have to test that everything is, in fact, working as advertised. I'd love nothing more than to be able to remove those notifications.

**Comment by pitiphong-p** [comment]
Well, I removed it and it work properly on my app.
Anyway you still have to remove observer returned from the addObserverForName:object:queue:usingBlock: right?

**Comment by mattt** [comment]
I not entirely sure about what's going on, but things don't work quite right without the notification stuff. Right now, the pattern looks like this:

``` objective-c
id observer = [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:childContext queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *note) {
    [context mergeChangesFromContextDidSaveNotification:note];
}];
//...
[[NSNotificationCenter defaultCenter] removeObserver:observer];
```

If I ever do figure out what's up, and it is indeed unnecessary, I'll gladly remove this.

**Pull request closed without merging**